wp-content/plugins/tapestry/controller/class.tapestry-controller.php

```diff
@@ -55,6 +52,73 @@ public function updateTapestry($tapestry) {
         return $tapestry;
     }
 
+    /**
+     * Retrieve a Tapestry post
+     * 
+     * @return Object Tapestry
+     */
+    public function getTapestry() {
+        if (!$this->postId) {
+            return $this->_throwsError('INVALID_POST_ID');
+        }
+        return $this->_getTapestryById($postId);
+    }
+
+    private function _getTapestryById($postId) {
+        $tapestry = get_post_meta($postId, 'tapestry', true);
+
+        $tapestry->nodes = array_map(function($nodeMetaId) {
+            $metadata = get_metadata_by_mid('post', $nodeMetaId);
+            $postId = $metadata->meta_value->post_id;
```

Comment:
Suggest you rename this to `$nodePostId` so the reader doesn't confuse it with the function parameter.